jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/azurerules/Subscription/RemoveCustomOwnerRoles.java

```diff
@@ -0,0 +1,160 @@
+                HashMultimap.create(), null, 0, new HashMap<>(), null, null);
+        logger.debug("Data fetched from elastic search. Response JSON: {}", resultJson.toString());
+
+        if (resultJson != null && resultJson.has(PacmanRuleConstants.HITS)) {
+            String hitsString = resultJson.get(PacmanRuleConstants.HITS).toString();
+            logger.debug("hit content in result json: {}", hitsString);
+            JsonObject hitsJson = JsonParser.parseString(hitsString).getAsJsonObject();
+            JsonArray hitsJsonArray = hitsJson.getAsJsonObject().get(PacmanRuleConstants.HITS).getAsJsonArray();
+            if (hitsJsonArray.size() > 0) {
+                JsonObject jsonDataItem = (JsonObject) ((JsonObject) hitsJsonArray.get(0))
+                        .get(PacmanRuleConstants.SOURCE);
+                logger.debug("Validating the data item: {}", jsonDataItem.toString());
+
+                String subscriptionId=jsonDataItem.getAsJsonObject().get("subscriptionId").getAsString();
+                logger.debug("Subscription id {}",subscriptionId);
+
+                JsonArray roleDefinitionList=jsonDataItem.getAsJsonObject().get("roleDefinitionList").getAsJsonArray();
+                logger.debug("roleDefinitionList size{}",roleDefinitionList);
+
+                if(!roleDefinitionList.isEmpty()){
+                    for(int i=0;i<roleDefinitionList.size();i++){
+                        JsonObject roleDefinitionObj=((JsonObject) roleDefinitionList.get(i));
+
+                        boolean actionFlag=false;
+                        JsonArray actions=roleDefinitionObj.getAsJsonObject().get("actions").getAsJsonArray();
+
+                        for(JsonElement action:actions){
+                            String actionValue=action.getAsString();
+                            logger.debug("actionValue {}",actionValue);
+
+                            if(actionValue.equalsIgnoreCase("*")) {
+                                actionFlag = true;
+                                break;
+                            }
+                        }
+
+                        boolean assignableScopesFlag=false;
+                        JsonArray assignableScopes=roleDefinitionObj.getAsJsonObject().get("assignableScopes").getAsJsonArray();
+
+                        for(JsonElement assignableScope:assignableScopes){
+                            String assignableScopeValue=assignableScope.getAsString();
+                            logger.debug("assignableScopes {}",assignableScopeValue);
+
+                            if(assignableScopeValue.equalsIgnoreCase("/")) {
+                                assignableScopesFlag = true;
+                                break;
+                            }
+                            else if (assignableScopeValue.equalsIgnoreCase("/subscriptions/"+subscriptionId)) {
+                                assignableScopesFlag=true;
+                                break;
+                            }
```

Comment:
These two if condtions can be combined with OR clause

Reply:
Done